pkg/types/azure/validation/platform.go

```diff
@@ -106,6 +110,15 @@ func validateAzureStack(p *azure.Platform, fldPath *field.Path) field.ErrorList
 	if p.ARMEndpoint == "" {
 		allErrs = append(allErrs, field.Required(fldPath.Child("armEndpoint"), "ARM endpoint must be set when installing on Azure Stack"))
 	}
+	if p.RHCOSImage == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("RHCOSImage"), "RHCOS Image must be set when installing on Azure Stack"))
+	} else {
+		storageBaseURL := strings.TrimPrefix(p.ARMEndpoint, "https://management.")
+		// If the URL for the image isn't in the ASH environment we can't use it.
+		if !strings.Contains(p.RHCOSImage, storageBaseURL) {
```

Comment:
I think we also need to address Matthew's original comment here 
https://github.com/openshift/installer/pull/5103/commits/f4de0c078b5fb4d8f01e76b94befcf34695a40a9#r725196893